src/main/java/net/tcpshield/tcpshield/fabric/impl/FabricPacketImpl.java

```diff
@@ -0,0 +1,24 @@
+package net.tcpshield.tcpshield.fabric.impl;
+
+import net.minecraft.network.packet.c2s.handshake.HandshakeC2SPacket;
+import net.tcpshield.tcpshield.abstraction.IPacket;
+import net.tcpshield.tcpshield.fabric.mixin.HandshakeC2SPacketAccessor;
+
+public class FabricPacketImpl implements IPacket {
+
+    private final HandshakeC2SPacket handshake;
+
+    public FabricPacketImpl(HandshakeC2SPacket handshake) {
+        this.handshake = handshake;
+    }
+
+    @Override
+    public String getRawPayload() {
+        return ((HandshakeC2SPacketAccessor) handshake).getAddress();
+    }
+
+    @Override
+    public void modifyOriginalPacket(String hostname) throws Exception {
+        // NO OPERATION
```

Comment:
Is the original hostname not stored on Fabric?

Reply:
As we modify the field where normally solely the server address / hostname goes on our servers,  `modifyOriginalPacket(String)` returns the packet to its original pre-TCPShield-proxy state. It's mostly for compatibility reasons such as `forceServer` (or similar) in BungeeCord, analytics on which subdomain the player joined, etc.

If Fabric doesn't store it and no popular mod accesses this packet field, I'm fine with the noop.